### PR TITLE
Throw error when resolver is not an object +  Better error messages when deserializing unknown nodes 

### PR DESCRIPTION
**Pull request by ankri**
Yay I'm back. With a smaller change this time.

This should help to prevent some common beginner mistakes.

### Display an error when `resolver` on `<Editor />` is not an object
image

### Display an error which component cannot be resolved
image

### CODE REVIEW

**Comment by ankri** [comment]
I replaced RTL with enzyme.  👍

File: packages/core/src/utils/deserializeNode.tsx [hunk 1]
```diff
@@ -71,6 +74,15 @@ export const deserializeNode = (
...
   const { type: Comp, props: Props, ...nodeData } = data;
 
+  invariant(
+    (Comp !== undefined && typeof Comp === 'string' && Comp.length > 0) ||
+      (Comp as { resolvedName?: string }).resolvedName !== undefined,
+    ERROR_DESERIALIZE_COMPONENT_NOT_IN_RESOLVER.replace(
+      '%displayName%',
+      data.displayName
+    ).replace('%availableComponents%', Object.keys(resolver).join(', '))
+  );
```
Comment (prevwong) [inline packages/core/src/utils/deserializeNode.tsx:84]: Can we break this line up so it is more readable? 

Ie:
```suggestion
   const isCompHtmlElement = Comp !== undefined && typeof Comp === 'string';
   const isCompUserComponent = Comp !== undefined && (Comp as { resolvedName?: string }).resolvedName !== undefined;

   invariant(isCompHtmlElement || isCompUserComponent,  '....')
```

**Review by prevwong — CHANGES_REQUESTED** [review]
Welcome back! Looks good, just 1 tiny suggestion!

File: packages/core/src/utils/deserializeNode.tsx (hunk 1 above)
Comment (ankri) [reply packages/core/src/utils/deserializeNode.tsx:84]: good idea, done

**Review by prevwong — APPROVED** [review]

**Pull request merged**